### PR TITLE
feat: add shopify-cart to nuxt reference store

### PR DESCRIPTION
**Pull request by KrisQ**
### WHY are these changes introduced?

Fixes [[ENG-7067](https://nacelle.atlassian.net/browse/ENG-7067)] 

### WHAT is this pull request doing?

Adds shopify cart to the nuxt reference store
Replace checkout with shopify cart checkout

### How to Test

- `git checkout ENG-7067--shopify-cart-nuxt-reference-store`
- to the nuxt reference store project `.env` add the following: (reach out if you want the vars I used)

```
SHOPIFY_SHOP_ID="<YOUR_SHOPIFY_ID>"
SHOPIFY_STOREFRONT_ACCESS_TOKEN="<YOUR_SHOPIFY_STOREFRONT_ACCESS_TOKEN>"
```

- `cd reference-stores/nuxt`
- `npm i && npm run generate && npm run start`
- see if cart/checkout works

### CODE REVIEW

**Review by NWRichmond — CHANGES_REQUESTED** [review]
## Dependencies & associated cleanup

Because we are replacing `@nacelle/shopify-checkout` with `@nacelle/shopify-cart` in this project, we should remove all artifacts of `@nacelle/shopify-checkout`. That includes:
- `.env.example`
- `nuxt.config.js`
- `package.json`

On the `package.json` front, we need to also add `@nacelle/shopify-cart`, and take care that `package-lock.json` is up-to-date. Here's what I did:

```
npm rm @nacelle/shopify-checkout

npm i @nacelle/shopify-cart

cd ../../

npm run bootstrap

cd -
```

## Optimistic UI

From the **Requirements** of [ENG-7067](https://nacelle.atlassian.net/browse/ENG-7067):

> Use [Optimistic UI](https://simonhearne.com/2021/optimistic-ui-patterns/#optimistic-cart-ui) to ensure that Shopify responses don't block cart functionality. For example, when adding an item to the cart, the item should be added instantly, with cartClient.cartLinesUpdate operating in the background. If prices depend on discounts, we can show loading UI on prices (including the subtotal) before we receive the cart response from Shopify.

I don't think that's happening in this project yet. There are blocking `await this.$cartClient.someMethod()` calls that result in the cart UI depending on Shopify Storefront API responses before Vuex and the cart UI is updated.

Instead, we should fire off those `this.$cartClient` methods in the background, and proceed with updating our stateful UI.

I added a comment in `store/cart.js` about the possibility of presenting a modal with error information in the event of `errors` or `userErrors`. Perhaps this is something that could be used any time a `this.$cartClient` method returns either `errors` or `userErrors`. That way, we're using Optimistic UI, but we're surfacing error information to avoid problematic mismatches between cart & checkout.

## Other updates

One thing to keep in mind is that there will probably need to be some changes made after a PR for [ENG-6877](https://nacelle.atlassian.net/browse/ENG-6877) is merged. I don't expect it'll be a big refactor, though. Just something to keep in mind, and should be accounted for before we merge this PR.

**Comment by KrisQ** [comment]
@NWRichmond Thank you for reviewing! I have fixed a lot and I am ready to refactor when `nacelleEntryId` becomes available

File: reference-stores/nuxt/store/cart.js [hunk 1]
```diff
@@ -2,10 +2,46 @@ import { get, set } from 'idb-keyval';
...
   lineItems: []
 });
 
+const handleResponseErrors = (response) => {
+  if (response.userErrors) {
+    console.error('User error', response.userErrors);
+  }
+  if (response.errors) {
+    console.error(response.errors);
+  }
```
[the remark this answers is not in the document]
Comment (NWRichmond) [reply reference-stores/nuxt/store/cart.js]: The new error modal is awesome! Thanks for this, @KrisQ.

Nuxt Reference Store error modal shopify cart

**Review by NWRichmond — CHANGES_REQUESTED** [review]
Nicely done, @KrisQ!

The new error modal is a very nice touch:
Nuxt Reference Store error modal shopify cart

(I produced an error by manipulating the `nacelleEntryId` in `this.$cartClient.cartLinesAdd`)

There are some lingering issues that I believe will be resolved by `@nacelle/shopify-cart` updates that are in-progress, with respect to `nacelleEntryId`s. So I think we're okay to ignore issues with those for now.

The only thing that seems missing is the dependency on `@nacelle/shopify-cart`. As requested in the previous review, I think we just need:

```
cd reference-stores/nuxt

npm i @nacelle/shopify-cart

cd ../../

npm run bootstrap
```

Once that's taken care of, I think this work is in a really good place.

**Comment by KrisQ** [comment]
Thank you @NWRichmond, the package should be installed correctly now!

**Pull request merged**